tests/Installer/Command/InstallCommandTest.php

```diff
@@ -0,0 +1,210 @@
+     * default config file is created during installation. It is not created when opted out via
+     * `--skip-database-config` command option.
+     *
+     * @test
+     */
+    public function installer_creates_db_config_file(): void
+    {
+        // Ensure that installer runs with no errors and creates system files as expected.
+        $systemFile = Config::locateConfigFile('system.yml');
+        $dbConfigFile = PIMCORE_APP_ROOT.'/config/local/database.yml';
+
+        $this->assertInstallerCommandRunsSuccessfully();
+
+        self::assertFileExists($systemFile);
+        self::assertFileExists($dbConfigFile);
+
+        // Ensure that local database config file is not created when opted out.
+        $this->filesystem->remove([$systemFile, $dbConfigFile]);
+
+        $this->assertInstallerCommandRunsSuccessfully(['--skip-database-config' => true]);
+
+        self::assertFileExists($systemFile);
+        self::assertFileNotExists($dbConfigFile);
+    }
+
+    protected function tearDown(): void
+    {
+        // Restore disposed system files.
+        array_map(
+            [$this, 'restoreFile'],
+            [
+                Config::locateConfigFile('system.yml'),
+                PIMCORE_APP_ROOT.'/config/local/database.yml',
+                PIMCORE_PROJECT_ROOT.'/vendor/pimcore/pimcore/bundles/CoreBundle/Migrations',
+            ]
+        );
+
+        parent::tearDown();
+    }
+
+    private function executeInstallerCliCommand(string $command, array $options = []): string
+    {
+        // @See bin/pimcore-install:50
+        $kernel = new InstallerKernel(PIMCORE_PROJECT_ROOT, Config::getEnvironment(), true);
+
+        $application = new Application($kernel);
+        $application->setAutoExit(false);
+
+        $applicationTester = new ApplicationTester($application);
+        $applicationTester->run(
+            array_merge(['command' => $command], $options),
+            ['capture_stderr_separately' => true]
+        );
+
+        return $applicationTester->getDisplay();
+    }
+
+    /**
+     * Remove a file keeping the ability to restore it with self::restoreFile()
+     */
+    private function disposeFile(string $filePath): void
+    {
+        $this->moveIfExits($filePath, $this->createBackupFileName($filePath));
+    }
+
+    /**
+     * Restores a file disposed by self::disposeFile()
+     *
+     * @See self::disposeFile()
+     */
+    private function restoreFile(string $filePath): void
+    {
+        $this->moveIfExits($this->createBackupFileName($filePath), $filePath);
+    }
+
+    /**
+     * Creates a symlink ($link) for a $target. If the link folder does not exist it will be
+     * recursively craeted.
+     * @param string $src
+     * @param string $dst
+     */
+    private function installMigrations(string $src, string $dst): void
+    {
+        $this->disposeFile($dst);
+        $this->filesystem->mkdir(dirname($dst));
```

Comment:
I'm not sure this is necessary, as `mirror()` internally calls [`$this->mkdir($targetDir)`](https://github.com/symfony/filesystem/blob/4.4/Filesystem.php#L564).

Reply:
good catch, @jdreesen !